### PR TITLE
allow bus_name attribute to be read. resolved bug in GenName().

### PR DESCRIPTION
**Pull request by allisonmcampbell**
Changes allow the bus_name attribute in the matpower case file to be read and turned into a dataframe accessible by the attribute bus_string.  Resolved bug in GenName() descriptor. Created unit test to confirm bus names/string are read as expected.

### CODE REVIEW

File: psst/case/__init__.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import pandas as pd
 import numpy as np
 
-from .descriptors import Name, Version, BaseMVA, BusName, Bus, Branch, BranchName, Gen, GenName, GenCost, GenType, GenFuel, Load, Period, _Attributes
+from .descriptors import Name, Version, BaseMVA, BusName, Bus, BusString, Branch, BranchName, Gen, GenName, GenCost, GenType, GenFuel, Load, Period, _Attributes
```
Comment (kdheepak) [inline psst/case/__init__.py:12]: `BusString` doesn't appear to be used anywhere?

File: psst/case/descriptors.py [hunk 2]
```diff
@@ -96,6 +96,10 @@ class Bus(Descriptor):
     name = 'bus'
     ty = pd.DataFrame
 
+class BusString(Descriptor):
```
Comment (allisonmcampbell) [inline psst/case/descriptors.py:99]: BusString used here

File: psst/case/descriptors.py [hunk 3]
```diff
@@ -124,6 +128,10 @@ def setattributeindex(self, instance, value):
             instance.load = pd.DataFrame(0, index=range(0, 1), columns=value, dtype='float')
 
         instance.bus.index = value
+        try:
+            instance.bus_string.index = value
```
Comment (allisonmcampbell) [inline psst/case/descriptors.py:132]: BusString used here

File: psst/case/__init__.py [hunk 4]
```diff
@@ -97,6 +97,8 @@ def _read_matpower(cls, mpc, auto_assign_names=True, fill_loads=True, remove_emp
 
                     if attribute == 'bus':
                         df.set_index('BUS_I',inplace=True)
+                    if attribute == 'bus_name':
```
Comment (allisonmcampbell) [inline psst/case/__init__.py:100]: BusString used here

File: psst/case/__init__.py (hunk 1 above)
Comment (allisonmcampbell) [reply psst/case/__init__.py:12]: see locations identified below

File: psst/case/descriptors.py (hunk 2 above)
Comment (kdheepak) [reply psst/case/descriptors.py:99]: The `BusString` class is defined here but no field is set as the `BusString` descriptor.

https://github.com/ames-market/psst/blob/817c2d15443e2ef901ba56aa7607bcbc0e654663/psst/case/__init__.py#L25-L38

Somewhere in the above lines, you'll need to add `bus_string = BusString()`

Comment (allisonmcampbell) [reply psst/case/descriptors.py:99]: whoops. thanks!

**Pull request merged**